akka-actor-typed/src/main/scala/akka/actor/typed/ActorSystem.scala

```diff
@@ -211,15 +212,16 @@ object ActorSystem {
                                 executionContext: Option[ExecutionContext] = None): ActorSystem[T] = {
 
     Behavior.validateAsInitial(guardianBehavior)
+    require(Behavior.isAlive(guardianBehavior))
     val cl = classLoader.getOrElse(akka.actor.ActorSystem.findClassLoader())
     val appConfig = config.getOrElse(ConfigFactory.load(cl))
     val setup = ActorSystemSetup(BootstrapSetup(classLoader, config, executionContext))
     val untyped = new a.ActorSystemImpl(name, appConfig, cl, executionContext,
-      Some(PropsAdapter(() ⇒ guardianBehavior, guardianProps)), setup)
+      Some(PropsAdapter(() ⇒ guardianBehavior, guardianProps, delayStart = true)), setup)
     untyped.start()
 
-    val adapter: ActorSystemAdapter.AdapterExtension = ActorSystemAdapter.AdapterExtension(untyped)
-    adapter.adapter
+    untyped.guardian ! ActorAdapter.Start
```

Comment:
I wonder how sure we can be that `Start` will be the first message that arrives at the guardian? It seems we would guarantee that only from the thread that created the `ActorSystem`.